runtime/parachains/src/paras_inherent.rs

```diff
@@ -981,29 +1010,28 @@ fn limit_disputes<T: Config>(disputes: &mut MultiDisputeStatementSet, entropy: [
 			}
 		});
 
-		// Compute the statements length of all remote disputes
-		let d = remote_disputes.iter().map(|d| d.statements.len() as u32).collect::<Vec<u32>>();
+		if INCLUDE_REMOTE && !remaining_weight.is_zero() {
```

Comment:
This is the main logic change that needs close review

Reply:
This needs additional thought, if there is still weight, what's the rationale of throwing away remote disputes? We can defer this discussion, but we should make an explicit statement about the rationale.

Reply:
If I recall correctly the rationale we discussed was that remote disputes are the main type that can be created as spam. But on further thought I am gonna remove this since I as long as we are within weight limit we should still include even if they end up being spam so the validator crafting the malicious disputes has a better chance of getting slashed